onnxruntime/core/providers/cpu/nn/regex_full_match.cc

```diff
@@ -0,0 +1,36 @@
+#include "regex_full_match.h"
+#include "core/common/common.h"
+
+namespace onnxruntime {
+ONNX_CPU_OPERATOR_KERNEL(
+    RegexFullMatch,
+    20,
+    KernelDefBuilder()
+        .TypeConstraint("T1", DataTypeImpl::GetTensorType<std::string>())
+        .TypeConstraint("T2", DataTypeImpl::GetTensorType<bool>()),
+    RegexFullMatch);
+
+RegexFullMatch::RegexFullMatch(const OpKernelInfo& info) : OpKernel(info) {
+  ORT_ENFORCE(info.GetAttr<std::string>("pattern", &pattern_).IsOK());
+  ORT_ENFORCE(RE2(pattern_).ok(), "Invalid pattern: ", pattern_);
+}
+
+Status RegexFullMatch::Compute(OpKernelContext* context) const {
+  RE2 re(pattern_);
+  const auto* input_tensor = context->Input<Tensor>(0);
+  if (nullptr == input_tensor) {
+    return Status(common::ONNXRUNTIME, common::FAIL, "Input count mismatch");
+  }
+  auto* output_tensor = context->Output(0, input_tensor->Shape());
+  if (nullptr == output_tensor) {
+    return Status(common::ONNXRUNTIME, common::FAIL, "Output count mismatch");
+  }
+  const auto input_data = input_tensor->template DataAsSpan<std::string>();
+  auto output_data = output_tensor->template MutableDataAsSpan<bool>();
+  for (auto i = 0; i < input_tensor->Shape().Size(); ++i) {
```

Comment:
Should we parallelize?

Reply:
We could do, it looks like the RE2 object is thread safe as well. Are there any pointers on how to use the ORT threadpool?